app/src/main/java/com/github/ashutoshgngwr/noice/WakeUpTimerManager.kt

```diff
@@ -142,44 +142,27 @@ object WakeUpTimerManager {
   }
 
   fun getStringScheduleLeftTime(context: Context, diffHours: Int, diffMinutes: Int): String {
-    return when (diffHours) {
-      0 -> {
-        when (diffMinutes) {
-          0 -> context.getString(R.string.wake_up_timer_schedule_set_minute, diffMinutes)
-          1 -> context.getString(R.string.wake_up_timer_schedule_set_minute, diffMinutes)
-          else -> context.getString(R.string.wake_up_timer_schedule_set_minutes, diffMinutes)
-        }
-      }
-      1 -> {
-        when (diffMinutes) {
-          0 -> context.getString(R.string.wake_up_timer_schedule_set_hour, diffHours)
-          1 -> context.getString(
-            R.string.wake_up_timer_schedule_set_hour_minute,
-            diffHours,
-            diffMinutes
-          )
-          else -> context.getString(
-            R.string.wake_up_timer_schedule_set_hour_minutes,
-            diffHours,
-            diffMinutes
-          )
-        }
-      }
-      else -> {
-        when (diffMinutes) {
-          0 -> context.getString(R.string.wake_up_timer_schedule_set_hours, diffHours)
-          1 -> context.getString(
-            R.string.wake_up_timer_schedule_set_hours_minute,
-            diffHours,
-            diffMinutes
-          )
-          else -> context.getString(
-            R.string.wake_up_timer_schedule_set_hours_minutes,
-            diffHours,
-            diffMinutes
-          )
-        }
-      }
+    val startString = context.getString(R.string.wake_up_timer_schedule_set)
+
+    val stringHours = if (diffHours == 0) {
+      ""
+    } else {
+      context.resources.getQuantityString(R.plurals.hours, diffHours, diffHours)
+    }
+
+    val stringMinutes = if (diffMinutes == 0 && diffHours > 0) {
+      ""
+    } else if (diffMinutes == 0 && diffHours == 0) {
+      context.getString(R.string.wake_up_timer_schedule_set_0_minutes)
+    } else {
+      context.resources.getQuantityString(R.plurals.minutes, diffMinutes, diffMinutes)
+    }
+
+    return if (diffHours > 0 && diffMinutes > 0) {
+      val bridge = context.getString(R.string.wake_up_timer_schedule_set_bridge)
+      "$startString $stringHours $bridge $stringMinutes."
+    } else {
+      "$startString $stringHours$stringMinutes."
```

Comment:
This concatenation will work for English, but it may not work for other languages. e.g., in my native language, the translation would be

> अलार्म 1 घंटे और 20 मिनट में बजेगा।

which, if translated one word at a time, translates to

> Alarm 1 hour and 20 minutes in will-ring.

I'd suggest using the original suggestion I made. Leave a placeholder in the message string to insert pluralised versions of minute and hour strings. e.g.

> Alarm will go off in %1$s and %2$s.

And then you can format it with the hour and minutes plurals.

Bonus: you can also make your bridge a plural. e.g.

```xml
<plurals name="hours">
  <item quantity="zero"></item>
  <item quantity="zero"></item>
  <item quantity="other">and</item>
</plurals>
```

Reply:
You're right, I didn't think about that, I'll set the string format with placeholders. About making bridge plural I've tried that but had no luck to make it work, maybe I'm doing something wrong? For now I'll set the string like you told me.

> Alarm will go off in %1$s and %2$s.

Reply:
I already fixed it in my local copy. Working on the test-case. Will push it soon.